### DISCUSSION TITLE
Adding grid current

### DISCUSSION THREAD

**Discussion by jseppeli** [hultenvp/solis-sensor#323] [Ideas]
Hi,

Currently there is Solis AC current for each phase separately.
Now wondering that could also the Grid AC current be added in the future?

Solis API documentation seems to have this 
/v1/api/epmDetail
iAc1
iAc2
iAc3

If poosible to add this would increase the possibility of checking the phase specific consumptions when having an energy meter connected to the inverter.

Br,
Jarkko

**Comment by hultenvp** [comment]
Hi Jarkko,

I can add that in the future. In the current implementation it would be pretty cumbersome to do so, but I've been working on a more generic client library for SolisCloud access last winter and will integrate that one in the integration coming winter period (at least that's the idea ;-) ). It will make it fully configurable what sensors to expose. Only problem will be that it will be a non-backward compatible change in sensor names.

**Reply by jseppeli** [reply]
You plans sounds nice, waiting for the future release!

**Comment by artarus** [comment]
> Hi,
> 
> Currently there is Solis AC current for each phase separately. Now wondering that could also the Grid AC current be added in the future?
> 
> Solis API documentation seems to have this /v1/api/epmDetail iAc1 iAc2 iAc3
> 
> If poosible to add this would increase the possibility of checking the phase specific consumptions when having an energy meter connected to the inverter.
> 
> Br, Jarkko

Hello Jarkko,
Actually no point to adding it from EPM as it works only if you have external EPM (Solis-EPM-5G), i have tested on my system and all values returned as "0". For simple meter as Acrel and others. are already in /inverterDetails. The working keys are for grid current: iA, iB, iC and for grid voltages: uA, uB, uC. Can be easy added to configuration. Also a power value can be added for each phase apart active power, as some issue for now, and ticket was raised about it by solis...

Kind Regards,
Arturas.
